pkg/kubelet/pod_workers.go

```diff
@@ -1159,6 +1163,8 @@ func (p *podWorkers) startPodSync(podUID types.UID) (ctx context.Context, update
 		klog.V(4).InfoS("Pod cannot start ever", "pod", klog.KObj(update.Options.Pod), "podUID", podUID, "updateType", update.WorkType)
 		return ctx, update, canStart, canEverStart, true
 	case !canStart:
+		// this is the only path we don't start the pod, so we need to put the change back in pendingUpdate
```

Comment:
nit: It might be worth to note that we don't need to set back the pendingUpdate in the `!canEverStart` case because, if canEverStart is true, then the pod worker loop will exit and nothing is expected to read the `pendingUpdate`.